### DISCUSSION TITLE
CSF actions not passed in from storyshots

### DISCUSSION THREAD

**Discussion by javidjamae** [storybookjs/storybook#12694] [Help]
I'm defining story-level argTypes that define actions like this:

```
export default {
  argTypes: { onSubmit: { action: 'Form Submitted' } },
  title: 'MyComponent',
  component: MyComponent,
}

export const HappyPath = ( { onSubmit } ) => {
  return <MyComponent onSubmit={ onSubmit } />
}
```

__This all works fine within the Storybook UI, but onSubmit is undefined when my Storyshots tests are run.__

I console.log onSubmit right at the top of the story and see that onSubmit is undefined when I run the test, but defined in the browser console when I pull open the UI.

Here is my storyshots config:

```
import initStoryshots, { multiSnapshotWithOptions } from '@storybook/addon-storyshots'

initStoryshots( {
  integrityOptions: { cwd: __dirname },
  test: multiSnapshotWithOptions( {
    createNodeMock: element => document.createElement( element.type ),
  } ),
} )
```

Also worth mentioning (perhaps), I have a global decorator defined in my `.storybook/preview.js` file. 

Is there something I have to configure to get Storyshots to pick up the argTypes that I define for each story?

**Comment by shilman** [comment]
This might be a storyshots bug. I don't remember where we got to with updating storyshots to handle args.

**Comment by appsparkler** [comment]
I'm resolving this by passing `defaultProps` :

```javascript
// MyComponent.js
//...
MyComponent.defaultProps = {
  onSubmit: () => null
}
//...
```
